props/saber_BC_buttons.h

```diff
@@ -1149,6 +1149,52 @@ EFFECT(tr);         // for EFFECT_TRANSITION_SOUND, use with User Effects.
 EFFECT(mute);       // Notification before muted ignition to avoid confusion.
 EFFECT(mzoom);      // for Spam Blast enter/exit
 
+template<class SPEC>
+struct BCScrollPresetsMode : public SPEC::SteppedMode {
+  int steps_per_revolution() override {
+    return 12;  // adjust for sensitivity
+  }
+
+  void next() override {
+    beeper.Beep(0.05, 4000);
+    prop_next_preset();
+  }
+
+  void prev() override {
+    beeper.Beep(0.05, 3000);
+    prop_previous_preset();
+  }
+
+  void update() override {  // Overridden to substitute the tick sound
+  }
+
+  void exit() override {
+    PVLOG_NORMAL << "** Exit Scroll Presets\n";
+    beeper.Beep(0.05, 3000);
+    beeper.Silence(0.05);
+    beeper.Beep(0.05, 3000);
+    beeper.Silence(0.05);
+    beeper.Beep(0.10, 2000);
+    SPEC::SteppedMode::exit();
+  }
+
+  bool mode_Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
```

Comment:
SteppedMode inherits from SelectCancelMode, which provides standard select and cancel(exit) functionality.
Since you're not calling Event2 in the base class, none of that will work for you.
If that is intended, I suggest adding a comment that explains it.

Reply:
Yeah, they're just pure custom button controls for this mode. Added comment to explain. (let me know if you meant something different)